Proofreading MO Guide (#11605)
openvinotoolkit/openvino: docs/MO_DG/prepare_model/convert_model/tf_specific/Convert_lm_1b_From_Tensorflow.md

# Converting a TensorFlow Language Model on One Billion Word Benchmark {#openvino_docs_MO_DG_prepare_model_convert_model_tf_specific_Convert_lm_1b_From_Tensorflow}

> ## Downloading the Pretrained Language Model on One Billion Word Benchmark

Reviewer: ```suggestion
## Downloading the Pre-trained Language Model on One Billion Word Benchmark
```
following the accepted version (by the popularity metric)

> TensorFlow provides [a pretrained Language Model on One Billion Word Benchmark](https://github.com/tensorflow/models/tree/r2.3.0/research/lm_1b).

Reviewer 2: ```suggestion
TensorFlow provides a pretrained [Language Model on One Billion Word Benchmark](https://github.com/tensorflow/models/tree/r2.3.0/research/lm_1b).
```

![lm_1b model view](../../../img/lm_1b.png)

The frozen model still has two variables: *`Variable`* and *`Variable_1`*.
It means that the model keeps training those variables at each inference.

At the first inference of this graph, the variables are initialized by initial values.
After executing the *`lstm`* nodes, results of execution are assigned to these two variables.

With each inference of the *`lm_1b`* graph, *`lstm`* initial states data is taken from previous inference
from variables, and states of current inference of *`lstm`* is reassigned to the same variables.

It helps the model to remember the context of the words that it takes as input.

## Converting a TensorFlow Language Model on One Billion Word Benchmark to IR

Model Optimizer assumes that output model is for inference only.
Therefore, you should cut those variables off and resolve keeping cell and hidden states on application level.

> There is a certain limitation for the model conversion:
> - Original model cannot be reshaped, so you should keep original shapes.

Reviewer: ```suggestion
There is a certain limitation for the model conversion: the original model cannot be reshaped, so you should keep original shapes.
```
